### PR TITLE
Fix for association versioning with an optional belongs to

### LINKED ISSUE #75: association versioning with an optional belongs to not working
I have a model with an optional belongs to. The history of the associated belongs_to model throws an error when trying to access the history when the reference was nil.

Have a PR with tests. It might be too simple but the tests are all passing.

### PR DESCRIPTION
**Pull request by ankursethi-uscis**
Fixes #75

### CODE REVIEW

**Comment by ankursethi-uscis** [comment]
Travis CI build is failing for rubocop violations in an unmodified file model_generator_spec.rb

**Comment by charlie-wasp** [comment]
@ankursethi-uscis Hi 👋 Sorry didn't have time to dig into it. I'll try to do it soon. Thank you for your contribution 👍

File: spec/logidze/model_spec.rb [hunk 1]
```diff
@@ -373,13 +374,16 @@
       after(:all) { Logidze.associations_versioning = false }
 
       describe "belongs_to" do
+        it "returns nil if past record does not exist" do
+          expect(article.at(version: 1).user).to be_nil
+        end
```
Comment (charlie-wasp) [inline spec/logidze/model_spec.rb]: Add the newline after `end`, please

File: spec/logidze/model_spec.rb [hunk 2]
```diff
@@ -373,13 +374,16 @@
       after(:all) { Logidze.associations_versioning = false }
 
       describe "belongs_to" do
+        it "returns nil if past record does not exist" do
```
Comment (charlie-wasp) [inline spec/logidze/model_spec.rb]: I believe, this construction would look better:

```ruby
context "when the association wasn't set in the past version" do
  it "returns nil" do
    ...
  end
end
```

**Comment by charlie-wasp** [comment]
Didn't find out yet, what the problem with Travis is, locally rubocop doesn't find any offenses on the master branch 🤔

**Comment by palkan** [comment]
> @palkan what do you think?

LGTM.

>  locally rubocop doesn't find any offenses on the master branch

Probably, your local version is out of date; try to update it.

Could you also freeze it in master (to the latest minor one, i.e. `~> 0.57.0`) and fix the offenses?

**Comment by ankursethi-uscis** [comment]
@charlie-wasp Made changes to the spec file and I fixed the rubocop issue. It was indeed an older rubocop which was specified in the `.gemspec`. In the spec file I left the outer `context '#belongs_to'` because it also describes the subsequent tests. You can see if it makes sense.

**Comment by ankursethi-uscis** [comment]
I see that test is failing for Rails 4.2. Quite a pain since the behavior is different between 4.2 and 5. In 4.2 belongs_to are optional. So the gem code fix is ok, but the dummy test app would have to be different. I added a fix to the dummy app.

**Comment by ankursethi-uscis** [comment]
@charlie-wasp Please check this again it has all the changes which were requested

**Comment by charlie-wasp** [comment]
@ankursethi-uscis thank you, and sorry for the long delay :( Before this PR will be merged, can you please add changelog entry about this fix?

**Comment by ankursethi-uscis** [comment]
@charlie-wasp Done and I squashed a few commits

**Comment by charlie-wasp** [comment]
@ankursethi-uscis to properly display the link to your github account in the changelog, please, add the next line to the bottom of the file:

`[@ankursethi-uscis]: https://github.com/ankursethi-uscis`

**Comment by ankursethi-uscis** [comment]
@charlie-wasp done

**Comment by charlie-wasp** [comment]
@ankursethi-uscis merged! 🎉 Thank you for the contribution!

**Pull request merged**